main.tsp

```diff
@@ -1,278 +0,0 @@
-
-  /** User that the todo is assigned to */
-  ownedBy: User.id;
-
-  /** A longer description of the todo item in markdown format */
-  description: string;
-
-  /** The status of the todo item */
-  status: TodoItemStatus;
-
-  /** When the todo item was created. */
-  @visibility("read") createdAt: utcDateTime;
-
-  /** When the todo item was last updated */
-  @visibility("read") updatedAt: utcDateTime;
-
-  /** When the todo item was makred as completed */
-  @visibility("read") completedAt: utcDateTime;
-
-  // Want the read form to be normalized to TodoLabelRecord[], but can't
-  // https://github.com/microsoft/typespec/issues/2926
-  labels: TodoLabel[];
-}
-
-union TodoLabel {
-  string,
-  string[],
-  TodoLabelRecord,
-  TodoLabelRecord[],
-}
-
-model TodoLabelRecord {
-  name: string;
-
-  @format("^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$")
-  color?: string;
-}
-
-union TodoAttachment {
-  file: TodoFileAttachment,
-  url: TodoUrlAttachment,
-}
-
-model TodoUrlAttachment {
-  /** A description of the URL */
-  description: string;
-
-  /** The url */
-  url: url;
-}
-
-model TodoFileAttachment {
-  /** The todo item this is attached to */
-  todoItemId: TodoItem.id;
-
-  /** The file name of the attachment */
-  @maxLength(255)
-  filename: string;
-
-  /** The media type of the attachment */
-  mediaType: string;
-
-  /** The url where the attachment can be downloaded from */
-  url: url;
-}
-
-enum TodoItemStatus {
-  NotStarted,
-  InProgress,
-  Completed,
-}
-
-@error
-model Error {
-  /** A machine readable error code */
-  code: string;
-
-  /** A human readable message */
-  message: string;
-}
-
-namespace Users {
-  // would prefer to extend
-  // https://github.com/microsoft/typespec/issues/2922
-
-  model UserCreatedResponse {
-    ...User;
-    ...OkResponse;
-
-    /** The token to use to construct the validate email address url */
-    token: string;
-  }
-
-  /** The user already exists */
-  model UserExistsResponse extends Error {
-    ...ConflictResponse;
-  }
-
-  /** The user is invalid (e.g. forgot to enter email address) */
-  model InvalidUserResponse extends Error {
-    @statusCode statusCode: 422;
-  }
-
-  @route("/users")
-  @post
-  op create(
-    user: User,
-  ): UserCreatedResponse | UserExistsResponse | InvalidUserResponse;
-
-  @route("validate")
-  @get
-  op validate(@query token: string): OkResponse | InvalidUserResponse;
-
-  @route("login")
-  @post
-  op login(
-    username: User.username,
-    password: User.password,
-  ): OkResponse | UnauthorizedResponse;
-
-  @route("logout")
-  @get
-  op logout(): OkResponse;
-
-  /** Sends a reset token to the user's email address */
-  @route("forgot-password")
-  @post
-  op forgotPassword(email: User.email): OkResponse | NotFoundResponse;
-
-  @route("reset-password")
-  @get
-  op resetPassword(@query resetToken: string): OkResponse | NotFoundResponse;
-}
-
-@route("items")
-namespace TodoItems {
-  model PaginationControls {
```

Comment:
I would have made this into an alias. I guess you could make an argument that you want to give (enforce/strongly encourage) that emitters should give these properties a name. And maybe it is the python in me that would prefer it to not have a name.

Reply:
You will easily end up emitting more "dead" models (or APIs with more object instantiations) if you sprinkle too many models in there. It does feel somewhat important to me to avoid naming too many things with public-facing names. I not super-opinionated either, I'm just on the other side of this fence. In other words, this is one of these problems that can take forever to sort out because neither one of us have strong enough feelings about it to make a strong enough argument :)